### ISSUE TITLE
Some imports not detected?

### ISSUE THREAD

**Issue by aryzing** [import-js/eslint-plugin-import#3192]
Given a pretty vanilla config, some of the imports in ESLint's config file itself, `eslint.config.mjs` are not being detected properly:

```ts
import js from "@eslint/js";
import globals from "globals";
import tseslint from "typescript-eslint"; // ERROR: import/no-unresolved
import { defineConfig } from "eslint/config"; // ERROR: import/no-unresolved
import importPlugin from "eslint-plugin-import";
import eslintConfigPrettier from "eslint-config-prettier/flat";
```

<details><summary>Full config</summary>

```js
import js from "@eslint/js";
import globals from "globals";
import tseslint from "typescript-eslint";
import { defineConfig } from "eslint/config";
import importPlugin from "eslint-plugin-import";
import eslintConfigPrettier from "eslint-config-prettier/flat";

export default defineConfig([
  {
    files: ["**/*.{js,mjs,cjs,ts}"],
    plugins: { js },
    extends: ["js/recommended"],
  },
  {
    files: ["**/*.{js,mjs,cjs,ts}"],
    languageOptions: {
      globals: {
        // Bun supports just about them all.
        ...globals.browser,
        ...globals.nodeBuiltin,
      },
    },
  },
  tseslint.configs.recommended,
  importPlugin.flatConfigs.recommended,
  // {
  //   files: ["**/*.{js,mjs,cjs,ts}"],
  //   ...importPlugin.flatConfigs.recommended,
  //   settings: {
  //     "import/resolver": {
  //       typescript: true,
  //     },
  //   },
  // },
  {
    files: ["**/*.{ts,tsx}"],
    extends: [importPlugin.flatConfigs.typescript],
    settings: {
      "import/resolver": {
        typescript: {
          bun: true,
        },
      },
    },
  },
  eslintConfigPrettier,
  {
    rules: {
      // Typically accepted pattern for ignored variables in Typescript. See
      // https://typescript-eslint.io/rules/no-unused-vars#what-benefits-does-this-rule-have-over-typescript
      "@typescript-eslint/no-unused-vars": [
        "error",
        {
          args: "all",
          argsIgnorePattern: "^_",
          caughtErrors: "all",
          caughtErrorsIgnorePattern: "^_",
          destructuredArrayIgnorePattern: "^_",
          varsIgnorePattern: "^_",
          ignoreRestSiblings: true,
        },
      ],
    },
  },
]);
```
</details> 

While experimenting, noticed that the following change allowed ESLint to pick up the imports again:

```diff
-  importPlugin.flatConfigs.recommended,
+  {
+    files: ["**/*.{js,mjs,cjs,ts}"],
+    ...importPlugin.flatConfigs.recommended,
+    settings: {
+      "import/resolver": {
+        typescript: true,
+      },
+    },
+  },
```

Why are only some imports being picked up when an explicit resolver isn't set?
